### PR TITLE
(SV) KM plot derive priority from config or related clinical attribute

### PR DESCRIPTION
**Pull request by dippindots**
Fix https://github.com/cbioportal/cbioportal/issues/7830
Describe changes proposed in this pull request:
1. Revert https://github.com/cBioPortal/cbioportal-frontend/pull/3372.
2. Decide whether to show the survival plot by default: get reserved priority from `StudyViewConfig` or use the priority from related `STATUS` clinical attribute (e.g. `***_STATUS`), hide the chart if priority is 0.
3. Change the km plot chart name to `KM Plot: NAME` (e.g. `Overall Survival` -> `KM Plot: Overall Survival`).
image

### CODE REVIEW

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -3616,14 +3604,19 @@ export class StudyViewPageStore {
...
                     dataType: getChartMetaDataType(survivalPlot.id),
                     patientAttribute: true,
                     displayName: survivalPlot.title,
-                    priority: getDefaultPriorityByUniqueKey(survivalPlot.id),
+                    // use survival status attribute's priority as KM plot's priority for non-reserved plots
+                    priority:
+                        STUDY_VIEW_CONFIG.priority[survivalPlot.id] ||
+                        getPriorityByClinicalAttribute(
+                            survivalPlot.survivalStatusAttribute
+                        ),
```
Comment (dippindots) [inline src/pages/studyView/StudyViewPageStore.ts:3619]: Derive priority from `STUDY_VIEW_CONFIG` or related clinical attribute.

File: src/pages/studyView/StudyViewPageStore.ts [hunk 2]
```diff
@@ -5056,28 +5049,36 @@ export class StudyViewPageStore {
...
-    }
+    readonly survivalPlots = remoteData<SurvivalType[]>({
+        await: () => [
+            this.survivalClinicalAttributesPrefix,
+            this.clinicalAttributeIdToClinicalAttribute,
+        ],
+        invoke: async () => {
+            let survivalTypes: SurvivalType[] = this.survivalClinicalAttributesPrefix.result.map(
+                prefix => {
+                    let plotTitle = `KM Plot: ${prefix} Survival`;
```
Comment (dippindots) [inline src/pages/studyView/StudyViewPageStore.ts:5060]: Change title for `KM plot`.

File: src/pages/studyView/StudyViewPageStore.ts [hunk 3]
```diff
@@ -3584,20 +3582,10 @@ export class StudyViewPageStore {
...
-        // only filter out survival attributes when there are more than 4 types of survival attributes
-        const filteredClinicalAttributes =
-            _.entries(this.survivalClinicalAttributesPrefix.result).length > 4
-                ? _.filter(this.clinicalAttributes.result, attribute =>
-                      notSurvivalAttribute(
-                          this.survivalClinicalAttributesPrefix.result,
-                          attribute.clinicalAttributeId
-                      )
-                  )
-                : this.clinicalAttributes.result;
```
Comment (dippindots) [inline src/pages/studyView/StudyViewPageStore.ts:3596]: Do not hide survival related clinical attributes anymore.

**Review by zhx828 — APPROVED** [review]
LGTM

**Comment by Luke-Sikina** [comment]
TODO: Verify on genie before releasing.

**Review by leexgh — APPROVED** [review]

File: src/pages/resultsView/survival/SurvivalUtil.tsx [hunk 4]
```diff
@@ -59,11 +59,11 @@ export const survivalPlotTooltipxLabelWithEvent: {
     DSS: 'Time of Death',
 };
 
-export const plotsPriority: { [prefix: string]: number } = {
-    OS: 1,
-    DFS: 2,
-    PFS: 3,
-    DSS: 4,
+export const RESERVED_SURVIVAL_PLOT_PRIORITY: { [prefix: string]: number } = {
```
Comment (alisman) [inline src/pages/resultsView/survival/SurvivalUtil.tsx:62]: need a comment explaining concept of RESERVED

File: src/pages/studyView/StudyViewPageStore.ts (hunk 1 above)
Comment (alisman) [reply src/pages/studyView/StudyViewPageStore.ts:3620]: i'm surprised that config takes precedence here (over data). is that right?

File: src/pages/studyView/StudyViewPageStore.ts [hunk 5]
```diff
@@ -5056,28 +5050,46 @@ export class StudyViewPageStore {
...
+            this.survivalClinicalAttributesPrefix,
+            this.clinicalAttributeIdToClinicalAttribute,
+        ],
+        invoke: async () => {
+            let survivalTypes: SurvivalType[] = this.survivalClinicalAttributesPrefix.result.map(
+                prefix => {
+                    // title shoud starts with 'KM plot:'
+                    let plotTitle = 'KM Plot: ';
+                    plotTitle +=
+                        this.survivalDescriptions.result &&
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts:5064]: should this be awaited above?

File: src/pages/studyView/StudyViewPageStore.ts [hunk 6]
```diff
@@ -5056,28 +5050,46 @@ export class StudyViewPageStore {
...
+                        this.survivalDescriptions.result[prefix]
+                            ? generateStudyViewSurvivalPlotTitle(
+                                  this.survivalDescriptions.result![prefix][0]
+                                      .displayName
+                              )
+                            : `${prefix} Survival`;
+
+                    let survivalStatusAttribute = this
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts:5072]: const

File: src/pages/resultsView/survival/SurvivalUtil.tsx (hunk 4 above)
Comment (dippindots) [reply src/pages/resultsView/survival/SurvivalUtil.tsx:62]: Sure

File: src/pages/studyView/StudyViewPageStore.ts [hunk 7]
```diff
@@ -5296,7 +5308,7 @@ export class StudyViewPageStore {
 
     public async getSurvivalDownloadData(chartMeta: ChartMeta) {
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts:5309]: check this, but i don't think there's any asyncronous stuff happening inside this method, so prob doesn't need to by async

File: src/pages/studyView/StudyViewPageStore.ts (hunk 1 above)
Comment (dippindots) [reply src/pages/studyView/StudyViewPageStore.ts:3620]: Yes, it is. For example, in this file: https://github.com/cBioPortal/cbioportal-frontend/blob/master/src/config/serverConfigDefaults.ts#L122-L125. We have priority four types of `KM plot`. We want to show them in front of the other charts. We can just ignore the priority in the related `survival status attribute` (e.g. `OS_STATUS`).

**Review by alisman — APPROVED** [review]

**Pull request merged**